### PR TITLE
feat(shorebird_cli): call `flutter precache` when installing new flutter rev

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

This provides better visibility into Flutter artifact downloads and why builds take as long as they do.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

### CODE REVIEW

File: packages/shorebird_cli/lib/src/shorebird_flutter.dart [hunk 1]
```diff
@@ -62,7 +71,24 @@ class ShorebirdFlutter {
...
+        ['precache', ...precacheArgs],
+        workingDirectory: targetDirectory.path,
+        runInShell: true,
+      );
+      precacheProgress.complete();
+    } catch (_) {
+      precacheProgress.fail('Failed to precache Flutter $version');
+      logger.info(
+        '''This is not a critical error, but your next build make take longer than usual.''',
+      );
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/shorebird_flutter.dart:90]: minor but maybe we can consolidate these into a single log like:

```dart
precacheProgress.fail('Failed to precache Flutter $version -- the next build make take longer than usual. ');
```

File: packages/shorebird_cli/lib/src/shorebird_flutter.dart [hunk 2]
```diff
@@ -62,7 +71,24 @@ class ShorebirdFlutter {
       rethrow;
     }
 
-    installProgress.complete();
+    final precacheProgress = logger.progress(
+      'Running ${lightCyan.wrap('flutter precache')}',
+    );
+
+    try {
+      await process.run(
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/shorebird_flutter.dart:79]: What about initial install? Should we also do this as part of the install script for consistency?

**Review by felangel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/shorebird_flutter.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/shorebird_flutter.dart:79]: Good idea, filed https://github.com/shorebirdtech/shorebird/issues/2093

File: packages/shorebird_cli/lib/src/shorebird_flutter.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/shorebird_flutter.dart:90]: I wanted to have a part of the message that wasn't a "failure" or "error" message to head off support requests when/if this fails (just to be extra clear that the thing that went wrong isn't going to cause shorebird to not work)

**Pull request merged**